tests/Feature/ScoutElasticSearchServiceProviderTest.php

```diff
@@ -34,4 +34,27 @@ public function test_provides()
         $provider = new ElasticSearchServiceProvider($this->app);
         $this->assertEquals([Client::class], $provider->provides());
     }
+
+    public function test_config_with_username()
+    {
+        $this->app['config']->set('elasticsearch.host', 'http://localhost:9200');
+        $this->app['config']->set('elasticsearch.user', 'elastic');
+        $this->app['config']->set('elasticsearch.password', 'pass');
+        $provider = new ElasticSearchServiceProvider($this->app);
+        $this->assertEquals([Client::class], $provider->provides());
+        /** @var Client $client */
+        $client = $this->app[Client::class];
```

Comment:
I could not find correct wat how I can check the configurations correctly passing to the client yet :)